LamestWebserver/LamestWebserver/Core/Logger.cs

```diff
@@ -259,28 +342,70 @@ public void CrashAndBurn(string msg)
             Environment.Exit(-1);
         }
 
-        internal void Log(ELoggingLevel loggingLevel, string msg)
+        private void Log(ELoggingLevel loggingLevel, string msg)
         {
-            // TODO: Implement propper logging!
-
-            ServerHandler.LogMessage($"[{loggingLevel.ToString()} \\\\\\\\ {msg}]");
+            if (_currentOutputSource != EOutputSource.None && _multiStreamWriter != null && !_multiStreamWriter.IsDisposed)
+            {
+                using (_loggerWriteLock.LockWrite())
+                {
+                    _multiStreamWriter.WriteLine($"[{loggingLevel.ToString()} \\\\\\\\ {msg}]");
+                }
+            }
         }
 
         /// <summary>
         /// Closes and Flushes the Logger stream.
+        /// If you want to restart the stream manually use 'Restart'.
         /// </summary>
         public void Close()
         {
-            // TODO: Close and Flush stream!
-            DebugExcept(new NotImplementedException());
+            using (_loggerWriteLock.LockWrite())
+            {
+                _multiStreamWriter.Dispose();
+            }
         }
 
         /// <summary>
         /// Opens and initializes the steam to write to.
+        /// If you want to restart the stream manually use 'Restart'.
         /// </summary>
         protected void Open()
         {
-            DebugExcept(new NotImplementedException());
+            using (_loggerWriteLock.LockWrite())
+            {
+                if (_currentOutputSource != EOutputSource.None)
+                {
+                    _createMultiStreamWR();
+                }
+            }
+        }
+
+        /// <summary>
+        /// Atomic operation to close and open the Logger save by a mutex.
+        /// </summary>
+        public void RestartStream()
+        {
+            using (_loggerWriteLock.LockWrite())
+            {
+                Close();
+                Open();
+            }
+        }
+
+        private void _createMultiStreamWR()
+        {
+            List<Stream> streamsToApply = new List<Stream>();
+            streamsToApply.Add(File.Open(_currentFilePath, FileMode.Append, FileAccess.Write));
+
+            if ((_currentOutputSource & EOutputSource.Console) == EOutputSource.Console)
+                streamsToApply.Add(Console.OpenStandardOutput());
+
+            if ((_currentOutputSource & EOutputSource.File) == EOutputSource.File)
+                streamsToApply.AddRange(_streams);
```

Comment:
this should just add the `FileStream` and also in the next line say `streamsToApply.AddRange(_streams);` in every case.